src/game/state.py

```diff
@@ -20,12 +21,13 @@ def from_dictionary(self, initial_state_dictionary):
                 self.place_piece(key, val)
```

Comment:
This function name (from_dictionary) suppose that you get a new state from dictionary. Please return a new State from it or rename it